codalab/lib/cache.py

```diff
@@ -0,0 +1,25 @@
+try:
+    import cPickle as pickle
+except:
+   import pickle
+
+import redis
+
+redis_connection = None
+
+def init(redis_connection_pool):
+    global redis_connection
+    if not redis_connection:
+        redis_connection = redis.Redis(connection_pool=redis_connection_pool)
+
+def get_or_compute(namespace, key, f):
+    if redis_connection:
+        redis_key = namespace + str(key)
```

Comment:
can we put a '/' between namespace and key?  Also, should we make the caller ensure `key` is a str (worried about running `str` on things that shouldn't be `str`-ified)?